vault/quotas/quotas_rate_limit_test.go

```diff
@@ -191,7 +192,7 @@ func TestRateLimitQuota_Allow_WithBlock(t *testing.T) {
 
 		// Since blocking is enabled, each client should only have 'rate' successful
```

Comment:
Could I ask you to make another change while you're here?  My read of this test is: over a 5s period, every 2ms, it spawns 5 goroutines, each with a different (fake) ClientAddress.  In other words, a total of 5 different IPs are seen as generating one request every 2ms.  We want to ensure that those IPs will only get 17 successful requests each during the interval.

My concern is that this amounts to a total of 500 goroutines per IP per second, or 12500 goroutines.  Of course normally we won't have anywhere near that many, because they should complete quickly.  Still, I feel we could protect ourselves against having too many easily enough by making the waitgroup local to the inner for loop, so we never have more than 5 goroutines in-flight at any time.

Reply:
You mean, "run 5, wait for 5" every 2ms? I guess that should work.

Reply:
Done.